airbyte-oauth/src/main/java/io/airbyte/oauth/flows/TikTokMarketingOAuthFlow.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.oauth.flows;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.oauth.BaseOAuth2Flow;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Following docs from
+ * https://ads.tiktok.com/marketing_api/docs?id=1701890912382977&is_new_connect=0&is_new_user=0&rid=sta6fe2yww
+ */
+public class TikTokMarketingOAuthFlow extends BaseOAuth2Flow {
+
+  private static final String ACCESS_TOKEN_URL = "https://business-api.tiktok.com/open_api/v1.2/oauth2/access_token/";
+
+  protected String getClientIdUnsafe(final JsonNode oauthConfig) {
+    return getConfigValueUnsafe(oauthConfig, "app_id");
+  }
+
+  protected String getClientSecretUnsafe(final JsonNode oauthConfig) {
+    return getConfigValueUnsafe(oauthConfig, "secret");
+  }
+
+  public TikTokMarketingOAuthFlow(final ConfigRepository configRepository, final HttpClient httpClient) {
+    super(configRepository, httpClient);
+  }
+
+  @VisibleForTesting
+  public TikTokMarketingOAuthFlow(final ConfigRepository configRepository, final HttpClient httpClient, final Supplier<String> stateSupplier) {
+    super(configRepository, httpClient, stateSupplier);
+  }
+
+  @Override
+  protected String formatConsentUrl(final UUID definitionId,
+                                    final String appId,
+                                    final String redirectUrl,
+                                    final JsonNode inputOAuthConfiguration)
+      throws IOException {
+
+    final URIBuilder builder = new URIBuilder()
+        .setScheme("https")
+        .setHost("ads.tiktok.com")
+        .setPath("marketing_api/auth")
```

Comment:
why does this logic use "ads.tiktok.com" here but for loading of access tokens another one (business-api.tiktok.com)?
I remember that ads.tiktok.com is the old API domain. Please check

Reply:
As I can see in the [documentation](https://ads.tiktok.com/marketing_api/docs?rid=esb88xalm6m&id=1709207085043713), TikTok changed its hostname only for API (`ads.tiktok.com/open_api` -> `business-api.tiktok.com/open_api`). Also, there is an Advertiser authorization URL in TikTok [Apps](https://ads.tiktok.com/marketing_api/apps) Basic information, it starts from `ads.tiktok.com`.